Migrate to GitHub actions

byroot opened the pull request:
Closes: https://github.com/nahi/httpclient/pull/463

Big thanks to @yahonda for the base PR, and @davidcornu for helping me with some OpenSSL head scratchers.

- Added a few gems to the Gemfile, like `logger` and `ostruct` as they've been extracted on recent rubies.
- Improve the rubynltm monkey patch.
- Changed the default timeouts in the test suite to avoid CI hanging when something goes wrong.
- I regenerated all the certificates used by the test suite because OpenSSL v3 was rejecting them because their private key was too small for modern standards.
- Unfortunately I couldn't make the 2.5 CI work, the version of bundler that ships with it is terribly broken.

cc @takkanm

byroot commented:
> * Unfortunately I couldn't make the 2.5 CI work, the version of bundler that ships with it is terribly broken.

I found a fix for it, so I restored it.

takkanm commented:
@byroot Thank you. I'm reviewing this PR. 

> Added a few gems to the Gemfile, like logger and ostruct as they've been extracted on recent rubies.

Both gems need to install httpclient. I think it is better to add both gems to gemspec file instead of Gemfile. What do you think?

@yahonda Could you review it if possible?

byroot commented:
The gems are only used for testing, so Gemfile is fine.Le 13 févr. 2025 à 11:41, Mitsutaka Mimura ***@***.***> a écrit :﻿
  @byroot Thank you. I'm reviewing this PR.

Added a few gems to the Gemfile, like logger and ostruct as they've been extracted on recent rubies.

Both gems need to install httpclient. I think it is better to add both gems to gemspec file instead of Gemfile. What do you think?
@yahonda Could you review it if possible?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>



takkanm left a comment (nahi/httpclient#469)
@byroot Thank you. I'm reviewing this PR.

Added a few gems to the Gemfile, like logger and ostruct as they've been extracted on recent rubies.

Both gems need to install httpclient. I think it is better to add both gems to gemspec file instead of Gemfile. What do you think?
@yahonda Could you review it if possible?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

takkanm commented:
> The gems are only used for testing, so Gemfile is fine.

Got it.

PR looks good to me. I want to merge it.
@yahonda Do you have any concerns?

yahonda commented:
This pull request looks good to me. It sets up the CI environment, making it easier to review future pull requests.

takkanm commented:
@yahonda Thank you for your review. 

I'll merge it. @byroot Thank you for your cooperation.
